Kernel/Clause.hpp

```diff
@@ -73,12 +73,25 @@ class Clause
     SELECTED = 4u
   };
 
-  Clause(unsigned length,const Inference& inf);
 
+private:
+  Clause(Literal* const* lits, unsigned length, Inference inf);
   void* operator new(size_t,unsigned length);
+public:
   void operator delete(void* ptr,unsigned length);
 
-  static Clause* fromStack(const Stack<Literal*>& lits, const Inference& inf);
+  static Clause* fromArray(Literal*const* lits, unsigned size, Inference inf)
+  { return new(size) Clause(lits, size, std::move(inf)); }
+
+  static Clause* fromLiterals(std::initializer_list<Literal*> lits, Inference inf)
+  { return fromArray(std::data(lits), lits.size(), std::move(inf)); }
+
+  static Clause* empty(Inference inf)
+  { return fromLiterals({}, inf); }
```

Comment:
Good question.. I think it doesn't matter., as we shouldn't access it because in this case `lits.size()` will be 0. 
Or is there anything I'm missing here?

Reply:
Yes, you're right - my concern was that `std::data({})` was UB, but apparently one possible implementation called `std::initializer_list<T>::begin` - which on an empty initializer list is "unspecified but identical to the `end()`" - so we're fine.